### ISSUE TITLE
resize issue

### ISSUE THREAD

**Issue by xuchengggg** [dbolya/yolact#124]
Hi, I saw two ways of resize in your code, one is keep the ratio and the other is just resize to 550x550. I want to know if you have trained both ways. I have trained with those two ways in tensorflow, and I got some interesting phenomenon. I use the first way to indicate the way to keep the image ratio as in faster_rcnn, and second way to indicate the other.

Firstly, in the training time, the first way can always be faster than second,  650ms/step of the first way and 860ms/step of the second trained on RTX2080Ti with batchsize of 8. And secondly, the mask_loss of first one will be higher while lower loc_loss and class_loss. When in eval time, the second way can have better mask, but a little lower confidence of objects.   Are those situations correct?

**Comment by xuchengggg** [comment]
In my code , keep aspect ratio one is faster to train, but I haven't to train test the eval time. I'm not sure is there any problem with my code, since I found it's really slow to train 550x550 without keep aspect ratio when i did not use random crop. 
And I found that in some test pictures, the one keep aspect ratio can always have higher confidence, but worse mask. I can pass you a few pictures to show that results. But I use resnet50 and trained it for 400000 steps,  it may not be fully trained.

**Comment by xuchengggg** [comment]
![微信图片_20190827195017](https://user-images.githubusercontent.com/40674552/63769098-80d4e980-c904-11e9-9e03-a744627a1b83.png)
![微信图片_20190827195007](https://user-images.githubusercontent.com/40674552/63769106-84687080-c904-11e9-8a65-0fca0edbd2df.png)

![微信图片_20190827195030](https://user-images.githubusercontent.com/40674552/63769116-8af6e800-c904-11e9-8134-2e90047f2863.png)
![微信图片_20190827195022](https://user-images.githubusercontent.com/40674552/63769122-8c281500-c904-11e9-8b3c-0bac082ec2cc.png)

First two are the results from your model with weights yolact_base_54_800000.pth, and the other are
the results from my code which resize inputs keep the aspect ratio.

**Comment by shoutOutYangJie** [comment]
hello， where does your tensorflow code from?

**Comment by xuchengggg** [comment]
> hello， where does your tensorflow code from?

By myself, but still have not got good results.

**Comment by xuchengggg** [comment]
I use the same strategy as mask_rcnn, and set the max_size to 550, min_size to 400, and then padded to 550x550. 
For just resize,  firstly
`image=dataset.load_image(image_id)`
`mask, class_ids=dataset.load_mask(image_id)`
and then use flip augmentation,  and get bbox by mask
`bbox = extract_bboxes(mask)`
and finally use the SSDAugmentation in your code, but just use random_crop and resize. Is there any problem?

And I suspect that there is a possibility that images input by the method of keep aspect ratio are filled with 0. For some kernels, the computational complexity is reduced, and the size of the input is reduced in disguise. This is just guess.

**Comment by xuchengggg** [comment]
OK. I will test the mAP next. But as you can see, the pictures above, the results of my implementation have a serious leakage. There may be those reasoons:
(1) I use resnet50, maybe resnet101 can get better results.
(2) For resize the image keep aspect ratio, in eval time, when get the final mask with 138x138, it still need to do the crop, and then resize to 550x550, maybe this would cause loss of information?
(3) There are some thing wrong in my code.
I really don't how to fix this problem.

**Comment by xuchengggg** [comment]
The third and fourth. I use the pre_trained weights of resnet50 and then train all the layers. Do I need to adjust my training strategy? Or is there some tricks I didn't notice?

**Comment by xuchengggg** [comment]
Thank you, I will try to convert pytorch weights to tensorflow. 
And you mean I should freeze all the batch norm layers when load resnet 50 pretrained weights?

**Comment by xuchengggg** [comment]
Ok, thanks, I will check my code then.

**Comment by sdimantsd** [comment]
> > hello， where does your tensorflow code from?
> 
> By myself, but still have not got good results.

Sup?
Anything new with the Yolact on tensorflow?

**Comment by Shubham3101** [comment]
@xuchengggg 
Hi, Can you share your implementation(tensorflow) of yolact?
